spec/commands/exec_spec.rb

```diff
@@ -6,6 +6,16 @@
     system_gems(system_gems_to_install, :path => :bundle_path)
   end
 
+  it "works with --gemfile flag", :focus do
+    create_file "CustomGemfile", <<-G
+      gem "rack"
+    G
+
+    bundle "exec --gemfile CustomGemfile test"
+
+    expect(ENV["BUNDLE_GEMFILE"]).to eq("CustomGemfile")
```

Comment:
I dont think this will test what you want it to, because the `bundle` helper actually runs bundler in a subprocess